>From f4b3b5b92aa37e8926a593ef1dc19965d553c288 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Ond=C5=99ej=20Sur=C3=BD?= <ondrej@sury.org>
Date: Thu, 10 Nov 2016 10:06:23 +0100
Subject: [PATCH 3/4] Add {pre,post,renew} hooks with 'run-parts
 /etc/letsencrypt/{pre,post,renew}-hook.d/'

---
 debian/certbot.cron.d  | 2 +-
 debian/certbot.dirs    | 3 +++
 debian/certbot.service | 2 +-
 3 files changed, 5 insertions(+), 2 deletions(-)
 create mode 100644 debian/certbot.dirs

diff --git a/debian/certbot.cron.d b/debian/certbot.cron.d
index dc2f28b..8da70cc 100644
--- a/debian/certbot.cron.d
+++ b/debian/certbot.cron.d
@@ -8,4 +8,4 @@
 SHELL=/bin/sh
 PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
 
-0 */12 * * * root test -x /usr/bin/certbot -a \! -d /run/systemd/system && perl -e 'sleep int(rand(3600))' && certbot -q renew
+0 */12 * * * root test -x /usr/bin/certbot -a \! -d /run/systemd/system && perl -e 'sleep int(rand(3600))' && certbot -q renew --pre-hook '/bin/run-parts /etc/letsencrypt/pre-hook.d/' --post-hook '/bin/run-parts /etc/letsencrypt/post-hook.d/' --renew-hook '/bin/run-parts /etc/letsencrypt/renew-hook.d/'
diff --git a/debian/certbot.dirs b/debian/certbot.dirs
new file mode 100644
index 0000000..9b2a1e4
--- /dev/null
+++ b/debian/certbot.dirs
@@ -0,0 +1,3 @@
+/etc/letsencrypt/pre-hook.d/
+/etc/letsencrypt/post-hook.d/
+/etc/letsencrypt/renew-hook.d/
diff --git a/debian/certbot.service b/debian/certbot.service
index e7f143c..2f1363d 100644
--- a/debian/certbot.service
+++ b/debian/certbot.service
@@ -4,5 +4,5 @@ Documentation=file:///usr/share/doc/python-certbot-doc/html/index.html
 Documentation=https://letsencrypt.readthedocs.io/en/latest/
 [Service]
 Type=oneshot
-ExecStart=/usr/bin/certbot -q renew
+ExecStart=/usr/bin/certbot -q renew --pre-hook '/bin/run-parts /etc/letsencrypt/pre-hook.d/' --post-hook '/bin/run-parts /etc/letsencrypt/post-hook.d/' --renew-hook '/bin/run-parts /etc/letsencrypt/renew-hook.d/'
 PrivateTmp=true
-- 
2.1.4

